# HopPER: an adaptive model for probability estimation of influenza reassortment through host prediction

Background: Influenza reassortment, a mechanism where influenza viruses exchange their RNA segments by
co-infecting a single cell, has been implicated in several major pandemics since 19th century. Owing to the significant
impact on public health and social stability, great attention has been received on the identification of influenza
reassortment.
Methods: We proposed a novel computational method named HopPER (Host-prediction-based Probability
Estimation of Reassortment), that sturdily estimates reassortment probabilities through host tropism prediction using
147 new features generated from seven physicochemical properties of amino acids. We conducted the experiments
on a range of real and synthetic datasets and compared HopPER with several state-of-the-art methods.
Results: It is shown that 280 out of 318 candidate reassortants have been successfully identified. Additionally, not
only can HopPER be applied to complete genomes but its effectiveness on incomplete genomes is also
demonstrated. The analysis of evolutionary success of avian, human and swine viruses generated through
reassortment across different years using HopPER further revealed the reassortment history of the influenza viruses.
Conclusions: Our study presents a novel method for the prediction of influenza reassortment. We hope this method
could facilitate rapid reassortment detection and provide novel insights into the evolutionary patterns of influenza
viruses
